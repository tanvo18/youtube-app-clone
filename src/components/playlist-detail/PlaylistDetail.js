import React, { Component } from 'react';
import PlaylistDetailStyled from './PlaylistDetailStyled';
import { Link } from 'react-router-dom';
import Indicator from '../indicator/Indicator';
import PropTypes from 'prop-types';
class PlaylistDetail extends Component {
  componentDidMount() {
    this.props.requestPlaylistVideo(this.props.selectedPlaylistId);
  }

  render() {
    const { playlistVideos, isFetching } = this.props;
    
    return (
      <PlaylistDetailStyled>
        <ul className="playlist">
          {
            isFetching
              ? <Indicator />
              : <div>
                {
                  playlistVideos && playlistVideos.length > 0
                    ? playlistVideos.map((videoItem, index) => (

                      <li
                        key={index}
                        className="playlist-video-item"
                      >
                        <Link
                          className="video-link"
                          to={`/watch/${videoItem.contentDetails.videoId}`}
                        >
                          <div className="thumbnail-wrap">
                            {
                              videoItem.snippet.thumbnails
                                ? <img src={videoItem.snippet.thumbnails.default.url} alt="" className="thumbnail-img" />
                                : <img src="" alt="" className="thumbnail-img" />
                            }
                          </div>
                          <div className="playlist-description-wrap">
                            <p className="title">{videoItem.snippet.title}</p>
                            <p className="channel-title">{videoItem.snippet.channelTitle}</p>
                          </div>
                        </Link>
                      </li>
                    ))
                    : <p>There are no playlist</p>
                }
              </div>
          }

        </ul>
      </PlaylistDetailStyled>
    );
  }
}

PlaylistDetail.defaultProps = {
  selectedPlaylistId: '',
  playlistVideos: [],
  requestPlaylistVideo: () => { },
};

PlaylistDetail.propTypes = {
  selectedPlaylistId: PropTypes.string,
  playlistVideos: PropTypes.array,
  requestPlaylistVideo: PropTypes.func,
};

export default PlaylistDetail;