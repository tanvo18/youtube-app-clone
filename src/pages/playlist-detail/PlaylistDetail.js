import React, { Component } from 'react';
import PlaylistDetailContainer from '../../containers/playlist-detail//PlaylistDetailContainer';
import PropTypes from 'prop-types';

class PlaylistDetail extends Component {

  render() {
    const selectedId = this.props.match.params.id;

    return (
      <div>
        <PlaylistDetailContainer
          selectedPlaylistId={selectedId}
        />
      </div>
    );
  }
}

PlaylistDetail.defaultProps = {
  match: {
    params: {
      id: ''
    }
  }
};

PlaylistDetail.propTypes = {
  match: PropTypes.object,
};
export default PlaylistDetail;