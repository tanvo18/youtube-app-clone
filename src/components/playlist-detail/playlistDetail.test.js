import React from 'react';
import PlaylistDetail from './PlaylistDetail';
import { BrowserRouter as Router } from 'react-router-dom';
import renderer from 'react-test-renderer';

const props = {
  playlistVideos: [
    {
      'id': 'PLYD4txFCoRIddpm1qaxj39HW0odA4Ojie',
      'snippet': {
        'publishedAt': '2018-02-17T22:36:32.000Z',
        'channelId': 'UCM6UtOuKhYDX2Q76rz_-Zug',
        'title': 'Football',
        'description': '',
        'thumbnails': {
          'default': {
            'url': 'https://i.ytimg.com/vi/YGpK6U56oHM/default.jpg',
            'width': 120,
            'height': 90
          },
        },
        'channelTitle': 'Tan Vo',
      },
      'contentDetails': {
        'videoId': 'FLqvTE1Eqfg',
      }
    },
  ]
};

const emptyPlaylistVideo = {
  playlistVideos: []
};

const propsEmptyThumbnail = {
  playlistVideos: [
    {
      'id': 'PLYD4txFCoRIddpm1qaxj39HW0odA4Ojie',
      'snippet': {
        'publishedAt': '2018-02-17T22:36:32.000Z',
        'channelId': 'UCM6UtOuKhYDX2Q76rz_-Zug',
        'title': 'Football',
        'description': '',
        'thumbnails': undefined,
        'channelTitle': 'Tan Vo',
      },
      'contentDetails': {
        'videoId': 'FLqvTE1Eqfg',
      }
    },
  ]
};

describe('test playlistDetail', () => {

  const requestPlaylistVideo = jest.fn();

  it('test requestPlaylistVideo function works', () => {
    const wrapper = shallow(<PlaylistDetail 
      requestPlaylistVideo={requestPlaylistVideo}
      {...props}
    />);
    wrapper.instance().componentDidMount();
    expect(requestPlaylistVideo).toHaveBeenCalled();
  });

  it('test empty playlistVideo data', () => {
    const wrapper = shallow(<PlaylistDetail 
      requestPlaylistVideo={requestPlaylistVideo}
      isFetching={false}
      {...emptyPlaylistVideo}
    />);
    expect(wrapper.find('.playlist').text()).toBe('There are no playlist');
  });

  it('test indicator works', () => {
    const wrapper = shallow(<PlaylistDetail 
      requestPlaylistVideo={requestPlaylistVideo}
      isFetching={true}
      {...emptyPlaylistVideo}
    />);
    expect(wrapper.find('Indicator').exists()).toBe(true);
    expect(wrapper.find('.playlist-video-item').exists()).toBe(false);
  });

  it('test indicator disappear when we have data', () => {
    const wrapper = shallow(<PlaylistDetail 
      requestPlaylistVideo={requestPlaylistVideo}
      isFetching={false}
      {...props}
    />);
    expect(wrapper.find('Indicator').exists()).toBe(false);
    expect(wrapper.find('.playlist-video-item').exists()).toBe(true);
  });

  it('empty thumbnail', () => {
    const wrapper = shallow(<PlaylistDetail 
      requestPlaylistVideo={requestPlaylistVideo}
      {...propsEmptyThumbnail}
    />);

    expect(wrapper.find('.thumbnail-img').exists()).toBe(true);
  });

  it('should render correctly', () => {
    const tree = renderer
      .create(<Router><PlaylistDetail 
        requestPlaylistVideo={requestPlaylistVideo}
        {...props}
      /></Router>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
  
});