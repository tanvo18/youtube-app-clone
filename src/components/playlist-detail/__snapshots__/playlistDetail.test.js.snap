// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test playlistDetail should render correctly 1`] = `
<div
  className="sc-bdVaJa hBAhhW"
>
  <ul
    className="playlist"
  >
    <div>
      <li
        className="playlist-video-item"
      >
        <a
          className="video-link"
          href="/watch/FLqvTE1Eqfg"
          onClick={[Function]}
        >
          <div
            className="thumbnail-wrap"
          >
            <img
              alt=""
              className="thumbnail-img"
              src="https://i.ytimg.com/vi/YGpK6U56oHM/default.jpg"
            />
          </div>
          <div
            className="playlist-description-wrap"
          >
            <p
              className="title"
            >
              Football
            </p>
            <p
              className="channel-title"
            >
              Tan Vo
            </p>
          </div>
        </a>
      </li>
    </div>
  </ul>
</div>
`;
